### PR TITLE
[KB] Match comments with generic mode

### PR DESCRIPTION
**Pull request by armchairlinguist**
Matching comments with generic mode - a nice little feature that we will now advertise a bit better. :)

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team

### CODE REVIEW

File: docs/kb/rules/match-commments.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+description: Semgrep's generic pattern matching mode can match comments in code files.
+tags:
+  - Rules
+  - Semgrep Code
+---
+
+# Match comments with Semgrep
+
+When Semgrep rules target a language or languages, they do not match comments in the targeted code files. Comments are not part of the semantic and syntactic structure of the document, so they are ignored.
```
Comment (khorne3) [inline docs/kb/rules/match-commments.md:10]:
```suggestion
When Semgrep rules target languages, they do not match comments in the targeted code files. Comments are not part of the semantic and syntactic structure of the document, so they are ignored.
```

File: docs/kb/rules/match-commments.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+However, it's sometimes useful to match comments. For example, comments can control the behavior of other linters, such as type checkers. You might also have certain formatting standards for comments, such as requiring that a `TODO` comment contains a ticket capturing the required work.
+
+To match comments with Semgrep, use the `generic` language target to invoke [generic pattern matching](/docs/writing-rules/generic-pattern-matching/).
+
+## Example rule
+
+Suppose that your organization requires all `TODO` comments to have an associated Jira ticket. This rule finds TODO lines with no `atlassian.net` content. It would identify any lines not containing a Jira Cloud ticket link.
```
Comment (khorne3) [inline docs/kb/rules/match-commments.md:18]:
```suggestion
Suppose that your organization requires all `TODO` comments to have an associated Jira ticket. This rule finds TODO lines with no `atlassian.net` content and identifies any lines not containing a Jira Cloud ticket link.
```

File: docs/kb/rules/match-commments.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    severity: INFO
+    metadata:
+      category: best-practice
+```
+
+:::note
+Try this pattern in the [Semgrep Playground](https://semgrep.dev/playground/s/lBDRL). 
+:::
+
+This rule also includes the `generic_ellipsis_max_span` option, which [limits the ellipsis to matching on the same line](/docs/writing-rules/generic-pattern-matching/#handling-line-based-input), and prevents it from over-matching in this generic context.
```
Comment (khorne3) [inline docs/kb/rules/match-commments.md:40]:
```suggestion
This rule also includes the `generic_ellipsis_max_span` option, which [limits the ellipsis to matching on the same line](/docs/writing-rules/generic-pattern-matching/#handling-line-based-input) and prevents it from over-matching in this generic context.
```

**Review by khorne3 — APPROVED** [review]
Some very small nits, but looks good to me otherwise!

File: docs/kb/rules/match-commments.md (hunk 1 above)
Comment (armchairlinguist) [reply docs/kb/rules/match-commments.md:10]: good call that this is awkward - I think I might go with "target specific languages" to make clear the opposition to `generic`

Comment (armchairlinguist) [reply docs/kb/rules/match-commments.md:10]: maybe I can just suggest things to myself, instead of doing the proper git thing... 😂

Comment (armchairlinguist) [inline docs/kb/rules/match-commments.md:10]:
```suggestion
When Semgrep rules target specific languages, they do not match comments in the targeted code files. Comments are not part of the semantic and syntactic structure of the document, so they are ignored.
```

**Pull request merged**